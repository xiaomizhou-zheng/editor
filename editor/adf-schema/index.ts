//
// This file is used in development to resolve types for the package (because they live in `src/`)
// *without* needing to use the `types` `package.json` field.
//
// We don't want to use the `types` field, as we want to point it to the declaration file in the
// published package.
//
// This file should be npm-ignored, so that it doesn't exist in the published package, and resolution
// falls through to the `types` and `main` `package.json` fields.
//
export {
  AnnotationTypes,
  B100,
  B400,
  B50,
  B500,
  B75,
  G200,
  G300,
  G400,
  G50,
  G500,
  G75,
  N0,
  N20,
  N200,
  N30,
  N300,
  N40,
  N50,
  N500,
  N60,
  N80,
  N800,
  N90,
  P100,
  P300,
  P400,
  P50,
  P500,
  P75,
  PanelType,
  R100,
  R300,
  R400,
  R50,
  R500,
  R75,
  T100,
  T300,
  T50,
  T500,
  T75,
  Y200,
  Y400,
  Y50,
  Y500,
  Y75,
  acNameToEmoji,
  acShortcutToEmoji,
  alignment,
  alignmentPositionMap,
  annotation,
  blockCard,
  blockquote,
  bodiedExtension,
  breakout,
  bulletList,
  bulletListSelector,
  code,
  codeBlock,
  codeBlockToJSON,
  colorPalette,
  colorPaletteExtended,
  confluenceInlineComment,
  confluenceJiraIssue,
  confluenceUnsupportedBlock,
  confluenceUnsupportedInline,
  copyPrivateMediaAttributes,
  createSchema,
  dataConsumer,
  date,
  decisionItem,
  decisionList,
  decisionListSelector,
  doc,
  em,
  emoji,
  emojiIdToAcName,
  expand,
  expandToJSON,
  extension,
  generateUuid,
  getEmojiAcName,
  getLinkMatch,
  hardBreak,
  heading,
  hexToRgb,
  hexToRgba,
  image,
  indentation,
  inlineCard,
  inlineExtension,
  inlineNodes,
  isHex,
  isRgb,
  isSafeUrl,
  layoutColumn,
  layoutSection,
  link,
  linkify,
  linkifyMatch,
  linkToJSON,
  listItem,
  media,
  mediaGroup,
  mediaSingle,
  mediaSingleToJSON,
  mediaToJSON,
  mention,
  mentionToJSON,
  nestedExpand,
  normalizeHexColor,
  normalizeUrl,
  orderedList,
  orderedListSelector,
  panel,
  paragraph,
  placeholder,
  rgbToHex,
  rule,
  sanitizeNodes,
  getCellDomAttrs,
  status,
  strike,
  strong,
  subsup,
  table,
  tableBackgroundBorderColor,
  tableBackgroundColorNames,
  tableBackgroundColorPalette,
  tableCell,
  tableCellContentDomSelector,
  tableCellContentWrapperSelector,
  tableCellSelector,
  tableHeader,
  tableHeaderSelector,
  tablePrefixSelector,
  tableRow,
  tableToJSON,
  taskItem,
  taskList,
  taskListSelector,
  text,
  textColor,
  toJSONTableCell,
  toJSONTableHeader,
  typeAheadQuery,
  underline,
  unknownBlock,
  unsupportedBlock,
  unsupportedInline,
  unsupportedNodeTypesForMediaCards,
  uuid,
  buildAnnotationMarkDataAttributes,
  AnnotationMarkStates,
} from './src';
export type {
  AlignmentAttributes,
  AlignmentMarkDefinition,
  AnnotationMarkAttributes,
  AnnotationMarkDefinition,
  BlockCardDefinition,
  BlockContent,
  BlockQuoteDefinition,
  BodiedExtensionDefinition,
  BreakoutMarkAttrs,
  BreakoutMarkDefinition,
  BulletListDefinition,
  CardAttributes,
  CellAttributes,
  CodeBlockAttrs,
  CodeBlockBaseDefinition,
  CodeBlockDefinition,
  CodeBlockWithMarksDefinition,
  CodeDefinition,
  DataConsumerAttributes,
  DataConsumerDefinition,
  DataType,
  DateDefinition,
  DecisionItemDefinition,
  DecisionListDefinition,
  DocNode,
  EmDefinition,
  EmojiAttributes,
  EmojiDefinition,
  ExpandDefinition,
  ExtensionDefinition,
  ExtensionLayout,
  ExternalMediaAttributes,
  HardBreakDefinition,
  HeadingBaseDefinition,
  HeadingDefinition,
  HeadingWithAlignmentDefinition,
  HeadingWithIndentationDefinition,
  HeadingWithMarksDefinition,
  IndentationMarkAttributes,
  IndentationMarkDefinition,
  Inline,
  InlineAtomic,
  InlineCardDefinition,
  InlineCode,
  InlineExtensionDefinition,
  InlineFormattedText,
  InlineLinkText,
  LayoutColumnDefinition,
  LayoutSectionDefinition,
  LinkAttributes,
  LinkDefinition,
  ListItemArray,
  ListItemDefinition,
  MarksObject,
  Match,
  MediaADFAttrs,
  MediaAttributes,
  MediaBaseAttributes,
  MediaDefinition,
  MediaDisplayType,
  MediaGroupDefinition,
  MediaSingleDefinition,
  MediaType,
  MentionAttributes,
  MentionDefinition,
  MentionUserType,
  NameToEmoji,
  NestedExpandContent,
  NestedExpandDefinition,
  NoMark,
  OrderedListDefinition,
  PanelAttributes,
  PanelDefinition,
  ParagraphBaseDefinition,
  ParagraphDefinition,
  ParagraphWithAlignmentDefinition,
  ParagraphWithIndentationDefinition,
  ParagraphWithMarksDefinition,
  PlaceholderDefinition,
  RuleDefinition,
  StatusDefinition,
  StrikeDefinition,
  StrongDefinition,
  SubSupAttributes,
  SubSupDefinition,
  TableAttributes,
  TableCellDefinition,
  TableDefinition,
  TableHeaderDefinition,
  TableLayout,
  TableRowDefinition,
  TaskItemDefinition,
  TaskListContent,
  TaskListDefinition,
  TextColorAttributes,
  TextColorDefinition,
  TextDefinition,
  UnderlineDefinition,
  UrlType,
  AnnotationId,
} from './src';
