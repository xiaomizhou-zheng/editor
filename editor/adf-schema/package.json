{
  "name": "@atlaskit/adf-schema",
  "version": "24.0.2",
  "description": "Shared package that contains the ADF-schema (json) and ProseMirror node/mark specs",
  "publishConfig": {
    "registry": "https://registry.npmjs.org/"
  },
  "repository": "https://bitbucket.org/atlassian/atlassian-frontend",
  "author": "Atlassian Pty Ltd",
  "license": "Apache-2.0",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "module:es2019": "dist/es2019/index.js",
  "types": "dist/types/index.d.ts",
  "sideEffects": false,
  "atlaskit:src": "src/index.ts",
  "atlassian": {
    "team": "Editor",
    "inPublicMirror": true,
    "releaseModel": "scheduled"
  },
  "af:exports": {
    ".": "./src/index.ts",
    "./steps": "./src/steps.ts",
    "./schema": "./src/schema/index.ts",
    "./json-schema": "./src/json-schema/index.ts",
    "./url": "./src/utils/url.ts",
    "./schema-bitbucket": "./src/schema/bitbucket-schema.ts",
    "./schema-confluence": "./src/schema/confluence-schema.ts",
    "./schema-jira": "./src/schema/jira-schema.ts",
    "./schema-default": "./src/schema/default-schema.ts"
  },
  "scripts": {
    "build:json-schema": "mkdir -p dist && cp -r json-schema dist",
    "generate:json-schema": "mkdir -p json-schema/v1/ && yarn generate:json-schema:full && yarn generate:json-schema:stage-0",
    "generate:json-schema:stage-0": "json-schema-generator --outDir=./json-schema/v1/ --stage=0 ./src/schema/nodes/doc.ts",
    "generate:json-schema:full": "json-schema-generator --outDir=./json-schema/v1/ ./src/schema/nodes/doc.ts",
    "ak-postbuild": "yarn build:json-schema"
  },
  "dependencies": {
    "@atlaskit/codemod-utils": "^4.1.0",
    "@atlaskit/editor-tables": "^2.2.0",
    "@babel/runtime": "^7.0.0",
    "@types/linkify-it": "^2.0.4",
    "@types/prosemirror-model": "^1.11.0",
    "@types/prosemirror-state": "^1.2.0",
    "css-color-names": "0.0.4",
    "linkify-it": "^2.0.3",
    "memoize-one": "^6.0.0",
    "prosemirror-model": "1.14.3",
    "prosemirror-transform": "1.3.2"
  },
  "devDependencies": {
    "@atlaskit/adf-utils": "^17.1.0",
    "@atlaskit/editor-json-transformer": "^8.8.0",
    "@atlaskit/editor-test-helpers": "^17.2.0",
    "@atlaskit/json-schema-generator": "^3.3.0",
    "@atlassian/adf-sample": "^1.1.0",
    "@atlassian/atlassian-frontend-prettier-config-1.0.1": "npm:@atlassian/atlassian-frontend-prettier-config@1.0.1",
    "@types/prosemirror-history": "^1.0.1",
    "ajv": "^6.12.6",
    "jscodeshift": "^0.13.0",
    "json-schema-diff-validator": "^0.4.1",
    "prosemirror-history": "^1.1.3",
    "prosemirror-state": "1.3.4",
    "typescript": "4.5.5"
  },
  "techstack": {
    "@atlassian/frontend": {
      "circular-dependencies": [
        "file-level"
      ]
    },
    "@repo/internal": {
      "deprecation": "no-deprecated-imports",
      "styling": [
        "emotion"
      ]
    }
  },
  "prettier": "@atlassian/atlassian-frontend-prettier-config-1.0.1"
}
