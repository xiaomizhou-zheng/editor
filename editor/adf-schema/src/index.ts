export {
  PanelType,
  AnnotationTypes,
  alignment,
  alignmentPositionMap,
  annotation,
  blockCard,
  blockquote,
  bodiedExtension,
  breakout,
  bulletList,
  bulletListSelector,
  caption,
  code,
  codeBlock,
  codeBlockToJSON,
  colorPalette,
  colorPaletteExtended,
  confluenceInlineComment,
  confluenceJiraIssue,
  confluenceUnsupportedBlock,
  confluenceUnsupportedInline,
  copyPrivateMediaAttributes,
  createSchema,
  dataConsumer,
  dataConsumerToJSON,
  date,
  decisionItem,
  decisionList,
  decisionListSelector,
  doc,
  em,
  embedCard,
  emoji,
  expand,
  expandToJSON,
  extension,
  fragment,
  fragmentToJSON,
  hardBreak,
  heading,
  image,
  indentation,
  inlineCard,
  inlineExtension,
  inlineNodes,
  layoutColumn,
  layoutSection,
  link,
  linkToJSON,
  listItem,
  media,
  mediaGroup,
  mediaSingle,
  mediaSingleWithCaption,
  mediaInline,
  mediaSingleToJSON,
  mediaToJSON,
  mention,
  mentionToJSON,
  nestedExpand,
  orderedList,
  orderedListSelector,
  panel,
  paragraph,
  placeholder,
  rule,
  sanitizeNodes,
  getCellAttrs,
  getCellDomAttrs,
  status,
  strike,
  strong,
  subsup,
  table,
  tableBackgroundBorderColor,
  tableBackgroundColorNames,
  tableBackgroundColorPalette,
  tableCell,
  tableCellContentDomSelector,
  tableCellContentWrapperSelector,
  tableCellSelector,
  tableHeader,
  tableHeaderSelector,
  tablePrefixSelector,
  tableRow,
  tableToJSON,
  taskItem,
  taskList,
  taskListSelector,
  text,
  textColor,
  toJSONTableCell,
  toJSONTableHeader,
  typeAheadQuery,
  underline,
  unknownBlock,
  unsupportedBlock,
  unsupportedInline,
  unsupportedNodeTypesForMediaCards,
  buildAnnotationMarkDataAttributes,
  AnnotationMarkStates,
  unsupportedMark,
  unsupportedNodeAttribute,
} from './schema';
export type {
  AlignmentAttributes,
  AlignmentMarkDefinition,
  AnnotationMarkAttributes,
  AnnotationMarkDefinition,
  BlockCardDefinition,
  BlockContent,
  BlockQuoteDefinition,
  BodiedExtensionDefinition,
  BreakoutMarkAttrs,
  BreakoutMarkDefinition,
  BulletListDefinition,
  CaptionDefinition,
  CardAttributes,
  CellAttributes,
  CodeBlockAttrs,
  CodeBlockBaseDefinition,
  CodeBlockDefinition,
  CodeBlockWithMarksDefinition,
  CodeDefinition,
  DataConsumerAttributes,
  DataConsumerDefinition,
  DataType,
  DateDefinition,
  DecisionItemDefinition,
  DecisionListDefinition,
  DocNode,
  EmbedCardDefinition,
  EmbedCardAttributes,
  EmDefinition,
  EmojiAttributes,
  EmojiDefinition,
  ExpandDefinition,
  ExtensionDefinition,
  ExtensionLayout,
  ExternalMediaAttributes,
  FragmentAttributes,
  FragmentDefinition,
  HardBreakDefinition,
  HeadingBaseDefinition,
  HeadingDefinition,
  HeadingWithAlignmentDefinition,
  HeadingWithIndentationDefinition,
  HeadingWithMarksDefinition,
  IndentationMarkAttributes,
  IndentationMarkDefinition,
  Inline,
  InlineAtomic,
  InlineCardDefinition,
  InlineCode,
  InlineExtensionDefinition,
  InlineFormattedText,
  InlineLinkText,
  LayoutColumnDefinition,
  LayoutSectionDefinition,
  LayoutSectionFullDefinition,
  LayoutSectionWithSingleColumnDefinition,
  LinkAttributes,
  LinkDefinition,
  ListItemArray,
  ListItemDefinition,
  MarksObject,
  MediaADFAttrs,
  MediaAttributes,
  MediaInlineAttributes,
  MediaInlineDefinition,
  MediaBaseAttributes,
  MediaDefinition,
  MediaDisplayType,
  MediaGroupDefinition,
  MediaSingleDefinition,
  MediaType,
  MentionAttributes,
  MentionDefinition,
  MentionUserType,
  NestedExpandContent,
  NestedExpandDefinition,
  NoMark,
  NonNestableBlockContent,
  OrderedListDefinition,
  PanelAttributes,
  PanelDefinition,
  ParagraphBaseDefinition,
  ParagraphDefinition,
  ParagraphWithAlignmentDefinition,
  ParagraphWithIndentationDefinition,
  ParagraphWithMarksDefinition,
  PlaceholderDefinition,
  RuleDefinition,
  StatusDefinition,
  StrikeDefinition,
  StrongDefinition,
  SubSupAttributes,
  SubSupDefinition,
  TableAttributes,
  TableCellDefinition,
  TableDefinition,
  TableHeaderDefinition,
  TableLayout,
  TableRowDefinition,
  TaskItemDefinition,
  TaskListContent,
  TaskListDefinition,
  TextColorAttributes,
  TextColorDefinition,
  TextDefinition,
  UnderlineDefinition,
  UrlType,
  AnnotationId,
  RichMediaAttributes,
  RichMediaLayout,
  AnnotationDataAttributes,
  CellDomAttrs,
} from './schema';
export {
  B100,
  B400,
  B50,
  B500,
  B75,
  G200,
  G300,
  G400,
  G50,
  G500,
  G75,
  N0,
  N20,
  N200,
  N30,
  N300,
  N40,
  N50,
  N500,
  N60,
  N80,
  N800,
  N90,
  P100,
  P300,
  P400,
  P50,
  P500,
  P75,
  R100,
  R300,
  R400,
  R50,
  R500,
  R75,
  T100,
  T300,
  T50,
  T500,
  T75,
  Y200,
  Y400,
  Y50,
  Y500,
  Y75,
  acNameToEmoji,
  acShortcutToEmoji,
  emojiIdToAcName,
  generateUuid,
  getEmojiAcName,
  getLinkMatch,
  hexToRgb,
  hexToRgba,
  isHex,
  isRgb,
  isSafeUrl,
  linkify,
  linkifyMatch,
  normalizeHexColor,
  normalizeUrl,
  rgbToHex,
  uuid,
} from './utils';
export type { Match, NameToEmoji } from './utils';
