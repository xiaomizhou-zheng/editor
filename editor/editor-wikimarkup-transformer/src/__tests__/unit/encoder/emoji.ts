import { defaultSchema } from '@atlaskit/adf-schema/schema-default';
import WikiMarkupTransformer from '../../../index';

import { doc, emoji, p } from '@atlaskit/editor-test-helpers/doc-builder';

describe('ADF => WikiMarkup - Emoji', () => {
  const transformer = new WikiMarkupTransformer();

  test('should convert emoji node', () => {
    const node = doc(
      p('Hello ', emoji({ id: '1f603', shortName: ':smiley:', text: 'ðŸ˜ƒ' })()),
    )(defaultSchema);
    expect(transformer.encode(node)).toMatchSnapshot();
  });

  test('should convert custom emoji node', () => {
    const node = doc(
      p(
        'Hello ',
        emoji({ id: '1234567890', shortName: ':partying_face:', text: 'ðŸ˜ƒ' })(),
      ),
    )(defaultSchema);
    expect(transformer.encode(node)).toMatchSnapshot();
  });
});
