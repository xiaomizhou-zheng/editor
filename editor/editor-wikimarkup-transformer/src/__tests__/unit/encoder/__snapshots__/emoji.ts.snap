// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ADF => WikiMarkup - Emoji should convert custom emoji node 1`] = `"Hello :partying_face:"`;

exports[`ADF => WikiMarkup - Emoji should convert emoji node 1`] = `"Hello ðŸ˜ƒ"`;
